components/button-group/index.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+function ButtonGroup( { className, ...props } ) {
+	const classes = classnames( 'components-button-group', className );
+
+	return (
+		<div { ...props } className={ classes } role="group" />
```

Comment:
>A group should be used to form a logical collection of items with related functionality, such as children in a tree widget forming a collection of siblings in a hierarchy, or a collection of items having the same container in a directory. **However, when a group is used in the context of list, authors must limit its children to [listitem](http://www.w3.org/TR/wai-aria/roles#listitem) elements.** Group elements may be nested.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_group_role

Maybe we need to [`Children.map`](https://reactjs.org/docs/react-api.html#reactchildrenmap) to apply a wrapper to each child?

Reply:
Discussed in slack channel and markup seemed good as is.